src/Serilog.Sinks.Raygun/Sinks/Raygun/RaygunSink.cs

```diff
@@ -109,12 +113,37 @@ public void Emit(LogEvent logEvent)
                 : new RaygunErrorMessage()
                 {
                     ClassName = logEvent.MessageTemplate.Text,
-                    Message = logEvent.MessageTemplate.Text,     
+                    Message = logEvent.MessageTemplate.Text,
                     Data = logEvent.Properties.ToDictionary(k => k.Key, v => v.Value.ToString())
                 };
 
             // Add user when requested
-            if (!string.IsNullOrWhiteSpace(_userNameProperty) &&
+            if (!string.IsNullOrWhiteSpace(_userInfoProperty) &&
+                logEvent.Properties.ContainsKey(_userInfoProperty) &&
+                logEvent.Properties[_userInfoProperty] != null)
+            {
+                var userInfoStructure = logEvent.Properties[_userInfoProperty] as StructureValue;
+                if (userInfoStructure != null)
+                {
+                    raygunMessage.Details.User = BuildUserInformationFromStructureValue(userInfoStructure);
+                }
+                else
+                {
+                    var userInfoScalar = logEvent.Properties[_userInfoProperty] as ScalarValue;
+                    if (userInfoScalar != null && userInfoScalar.Value is string)
```

Comment:
Likewise, `if (userInfoProperty is ScalarValue userInfoScalar)` would work, here.